events-api/src/modules/events/events.controller.ts

```diff
@@ -21,5 +31,99 @@ EventsController.get(
       data: events,
       searchParams,
     });
-  }
+  },
+);
+
+EventsController.post(
+  "/",
+  validateRequestBody(eventsCreateRequestBodyDtoSchema),
+  async (req, res) => {
+    const newEvent = await EventsService.createEvent(
+      req.body as unknown as eventsCreateRequestBodyDto,
+    );
+
+    return res.status(201).json({
+      message: "Event created successfully",
+      data: newEvent,
+    });
+  },
+);
+
+EventsController.get(
+  "/:eventId",
+  validateIdRouteParameter("eventId"),
+  async (req, res) => {
+    const event = await EventsService.getEvent(req.params["eventId"] as string);
+
+    if (!event) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(200).json({
+      message: "Event retrieved successfully",
+      data: event,
+    });
+  },
+);
+
+EventsController.patch(
+  "/:eventId",
+  validateIdRouteParameter("eventId"),
+  validateRequestBody(eventsUpdateRequestBodyDtoSchema),
+  async (req, res) => {
+    const updatedEvent = await EventsService.updateEvent(
+      req.body as unknown as eventsUpdateRequestBodyDto,
+      req.params["eventId"] as string,
+    );
+
+    if (!updatedEvent) {
+      return res.status(404).json({ message: "Event not found" });
```

Comment:
Such kind of error handling is better to perform at the global error handler level